### DISCUSSION TITLE
Shoulder Surfing defenses (hiding PID's and such)

### DISCUSSION THREAD

**Discussion by ghost** [sandboxie-plus/Sandboxie#3698] [Q&A Feedback]
I think Sandboxie momentarily lists PID's when it launches a sandboxed process. Is it possible to disable that feature to improve shoulder surfing protection? 

This should probably be placed in "Ideas and Suggestions" category.

**Comment by offhub** [comment]
If you use the Sandboxie Plus interface, you can turn off the PID column.
If you use the Sandboxie Classic interface, you can resize the PID column.

**Comment by bastik-1001** [comment]
In the same way as offhub describes, within Sandman you can choose not to display the title and the path, if they consider those to be something an onlooker should not be able to see. The window title of a window under Sandboxie's control, contains potentially private information, that is displayed in the sandbox manager, while the window is not shown.

**Comment by ghost** [comment]
I knew about the columns, but I think the window I saw flashing for a second was a CMD window, not a GUI one.

**Comment by ghost** [comment]
Now I know for sure that Sandboxie briefly displays a window with PID's when a sandboxed application is launched. This shouldn't be happening. To an advanced attacker knowing PID and/or TID is almost as bad as knowing victim's password.

**Reply by offhub** [reply]
Sandboxie will not run any additional programs unless you configure it to do so. If you can provide some information, such as what application and configuration you were using, we may be able to help you.

**Reply by diversenok** [reply]
> To an advanced attacker knowing PID and/or TID is almost as bad as knowing victim's password.

This is total nonsense. PID and TID values give you nothing. They are easy to guess (as they are small multiples of 4 [allocated in sequential blocks](https://www.google.com/books/edition/Windows_Internals/y83LDgAAQBAJ?hl=en&gbpv=1&printsec=frontcover&bsq=a%20global%20handle%20table%20PspCidTable)); they are [possible to enumerate](https://github.com/diversenok/Things-AppContainer-Knows) in many different ways, even from the context of the strongest browser sandboxes; and they do not offer any advantages for gaining [more access than allowed](https://learn.microsoft.com/en-us/windows/win32/secauthz/securable-objects). The icons on your taskbar reveal many times more information to attackers then your PID values.

**Reply by bastik-1001** [reply]
> Now I know for sure that Sandboxie briefly displays a window with PID [...]
How did you get to know? Or in other words, how can it be reproduced?

> [....] knowing PID and/or TID is almost as bad as knowing victim's password.

Are you confusing it with a PIN? (many of the generated PIDs are surely someone's PIN) If not, what can be gained by knowing it?

**Reply by bastik-1001** [reply]
Well, today I learned about [Parent PID Spoofing](https://attack.mitre.org/techniques/T1134/004/). A lot [more background](https://pentestlab.blog/2020/02/24/parent-pid-spoofing/) on it. This is a Windows issue and anything started in the sandbox should stay in the sandbox, as far as Sandboxie is concerned.
